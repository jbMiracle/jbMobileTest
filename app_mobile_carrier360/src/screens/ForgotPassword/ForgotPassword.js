import React, { Component } from 'react';
import { Alert } from 'react-native';
import { connect } from 'react-redux';
import { connectStyle, Text, Button } from 'native-base';
import PropTypes from 'prop-types';
import styleDef from './ForgotPassword.style';
import strings from '../../services/localization';
import C360LogoContainer from '../../components/C360LogoContainer/C360LogoContainer';
import { App as AppActions } from '../../state/actions';
import C360EmailInputField from '../../components/C360EmailInputField/C360EmailInputField';

const forgotTextOptions = { scope: 'ForgotPassword' };
const mapStateToProps = state => ({
  forgotPasswordSuccess: state.forgotPassword.forgotPasswordSuccess,
  forgotPasswordError: state.forgotPassword.forgotPasswordError,
});
const mapDispatchToProps = dispatch => ({
  sendPasswordReset: email => dispatch(AppActions.forgotPassword(email)),
});

export class ForgotPasswordScreen extends Component {
  state = {
    hasValidEmail: false,
    emailAddress: '',
  };

  componentDidUpdate(prevProps) {
    if (prevProps.forgotPasswordSuccess !== this.props.forgotPasswordSuccess && this.props.forgotPasswordSuccess) {
      this.showSuccessMessage();
    } else if (prevProps.forgotPasswordError !== this.props.forgotPasswordError && this.props.forgotPasswordError) {
      this.showErrorMessage(this.props.forgotPasswordError);
    }
  }

  showSuccessMessage = () => {
    Alert.alert(
      null,
      strings('success', forgotTextOptions),
      [
        { text: strings('App.genericOk'), onPress: () => this.props.navigation.pop() },
      ],
    );
  };

  showErrorMessage = (message) => {
    Alert.alert(null, message);
  };

  requestPasswordReset = () => {
    this.props.sendPasswordReset(this.state.emailAddress);
  };

  render() {
    return (
      <C360LogoContainer
        showBackButton
        backButtonOnPress={ () => this.props.navigation.pop() }
        headerText={ strings('forgetTitle', forgotTextOptions) }
        descriptionText={ strings('forgetBody', forgotTextOptions) }
      >

        <C360EmailInputField
          style={ styleDef.inputStyle }
          placeHolderText={ strings('inputPlaceholder', forgotTextOptions) }
          value={ this.state.emailAddress }
          onChangeText={ emailAddress => this.setState(state => ({ ...state, emailAddress })) }
          onValidationUpdate={ hasValidEmail => this.setState(state => ({ ...state, hasValidEmail })) }
          leftIcon='jbh-Contact_Mail'
          dark
        />

        <Button
          testID='forgotPassword_requestButton'
          style={ styleDef.buttonStyle }
          onPress={ this.requestPasswordReset }
          block
          large
          primaryDark
          disabled={ !this.state.hasValidEmail }
        >
          <Text>{ strings('requestButton', forgotTextOptions) }</Text>
        </Button>
      </C360LogoContainer>
    );
  }
}

ForgotPasswordScreen.defaultProps = {
  forgotPasswordError: null,
  forgotPasswordSuccess: false,
};

ForgotPasswordScreen.propTypes = {
  forgotPasswordError: PropTypes.string,
  forgotPasswordSuccess: PropTypes.bool,
  navigation: PropTypes.shape({
    pop: PropTypes.func.isRequired,
  }).isRequired,
  sendPasswordReset: PropTypes.func.isRequired,
};

const ReduxConnectedForgotPassword = connect(mapStateToProps, mapDispatchToProps)(ForgotPasswordScreen);
export default connectStyle('C360.ForgotPassword', styleDef)(ReduxConnectedForgotPassword);
