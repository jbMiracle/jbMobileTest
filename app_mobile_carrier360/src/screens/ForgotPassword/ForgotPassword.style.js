export default {
  buttonStyle: {
    marginTop: 56,
  },
  inputStyle: {
    marginTop: 44,
  },
};
