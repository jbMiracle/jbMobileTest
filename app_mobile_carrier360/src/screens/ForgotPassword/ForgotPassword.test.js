import React from 'react';
import { shallow } from 'enzyme';
import { ForgotPasswordScreen } from './ForgotPassword';

const mockProps = {
  navigation: {
    pop: () => {},
  },
  sendPasswordReset: () => {},
};

describe('ForgotPassword Screen', () => {
  it('can render initially with a disabled button', () => {
    const wrapper = shallow(<ForgotPasswordScreen { ...mockProps } />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can render with a disabled button when it has an email', () => {
    const wrapper = shallow(<ForgotPasswordScreen { ...mockProps } />);
    wrapper.setState({
      hasValidEmail: true,
      emailAddress: 'test@test.com',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
