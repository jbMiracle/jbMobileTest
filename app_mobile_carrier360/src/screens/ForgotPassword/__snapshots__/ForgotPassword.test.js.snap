// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword Screen can render initially with a disabled button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPasswordScreen
    forgotPasswordError={null}
    forgotPasswordSuccess={false}
    navigation={
      Object {
        "pop": [Function],
      }
    }
    sendPasswordReset={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backButtonOnPress": [Function],
      "children": Array [
        <C360EmailInputField
          dark={true}
          leftIcon="jbh-Contact_Mail"
          onChangeText={[Function]}
          onValidationUpdate={[Function]}
          placeHolderText="Email address"
          style={
            Object {
              "marginTop": 44,
            }
          }
          value=""
        />,
        <Styled(Button)
          block={true}
          disabled={true}
          large={true}
          onPress={[Function]}
          primaryDark={true}
          style={
            Object {
              "marginTop": 56,
            }
          }
          testID="forgotPassword_requestButton"
          virtual={undefined}
        >
          <Styled(Text)
            virtual={undefined}
          >
            Request
          </Styled(Text)>
        </Styled(Button)>,
      ],
      "descriptionText": "Please enter the email address associated to your account to request a password reset. Check that email account for further instructions.",
      "headerText": "Forget your password?",
      "showBackButton": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dark": true,
          "leftIcon": "jbh-Contact_Mail",
          "onChangeText": [Function],
          "onValidationUpdate": [Function],
          "placeHolderText": "Email address",
          "style": Object {
            "marginTop": 44,
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": <Styled(Text)
            virtual={undefined}
          >
            Request
          </Styled(Text)>,
          "disabled": true,
          "large": true,
          "onPress": [Function],
          "primaryDark": true,
          "style": Object {
            "marginTop": 56,
          },
          "testID": "forgotPassword_requestButton",
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Request",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Request",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backButtonOnPress": [Function],
        "children": Array [
          <C360EmailInputField
            dark={true}
            leftIcon="jbh-Contact_Mail"
            onChangeText={[Function]}
            onValidationUpdate={[Function]}
            placeHolderText="Email address"
            style={
              Object {
                "marginTop": 44,
              }
            }
            value=""
          />,
          <Styled(Button)
            block={true}
            disabled={true}
            large={true}
            onPress={[Function]}
            primaryDark={true}
            style={
              Object {
                "marginTop": 56,
              }
            }
            testID="forgotPassword_requestButton"
            virtual={undefined}
          >
            <Styled(Text)
              virtual={undefined}
            >
              Request
            </Styled(Text)>
          </Styled(Button)>,
        ],
        "descriptionText": "Please enter the email address associated to your account to request a password reset. Check that email account for further instructions.",
        "headerText": "Forget your password?",
        "showBackButton": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dark": true,
            "leftIcon": "jbh-Contact_Mail",
            "onChangeText": [Function],
            "onValidationUpdate": [Function],
            "placeHolderText": "Email address",
            "style": Object {
              "marginTop": 44,
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": <Styled(Text)
              virtual={undefined}
            >
              Request
            </Styled(Text)>,
            "disabled": true,
            "large": true,
            "onPress": [Function],
            "primaryDark": true,
            "style": Object {
              "marginTop": 56,
            },
            "testID": "forgotPassword_requestButton",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Request",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Request",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ForgotPassword Screen can render with a disabled button when it has an email 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPasswordScreen
    forgotPasswordError={null}
    forgotPasswordSuccess={false}
    navigation={
      Object {
        "pop": [Function],
      }
    }
    sendPasswordReset={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backButtonOnPress": [Function],
      "children": Array [
        <C360EmailInputField
          dark={true}
          leftIcon="jbh-Contact_Mail"
          onChangeText={[Function]}
          onValidationUpdate={[Function]}
          placeHolderText="Email address"
          style={
            Object {
              "marginTop": 44,
            }
          }
          value="test@test.com"
        />,
        <Styled(Button)
          block={true}
          disabled={false}
          large={true}
          onPress={[Function]}
          primaryDark={true}
          style={
            Object {
              "marginTop": 56,
            }
          }
          testID="forgotPassword_requestButton"
          virtual={undefined}
        >
          <Styled(Text)
            virtual={undefined}
          >
            Request
          </Styled(Text)>
        </Styled(Button)>,
      ],
      "descriptionText": "Please enter the email address associated to your account to request a password reset. Check that email account for further instructions.",
      "headerText": "Forget your password?",
      "showBackButton": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dark": true,
          "leftIcon": "jbh-Contact_Mail",
          "onChangeText": [Function],
          "onValidationUpdate": [Function],
          "placeHolderText": "Email address",
          "style": Object {
            "marginTop": 44,
          },
          "value": "test@test.com",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": <Styled(Text)
            virtual={undefined}
          >
            Request
          </Styled(Text)>,
          "disabled": false,
          "large": true,
          "onPress": [Function],
          "primaryDark": true,
          "style": Object {
            "marginTop": 56,
          },
          "testID": "forgotPassword_requestButton",
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Request",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Request",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backButtonOnPress": [Function],
        "children": Array [
          <C360EmailInputField
            dark={true}
            leftIcon="jbh-Contact_Mail"
            onChangeText={[Function]}
            onValidationUpdate={[Function]}
            placeHolderText="Email address"
            style={
              Object {
                "marginTop": 44,
              }
            }
            value="test@test.com"
          />,
          <Styled(Button)
            block={true}
            disabled={false}
            large={true}
            onPress={[Function]}
            primaryDark={true}
            style={
              Object {
                "marginTop": 56,
              }
            }
            testID="forgotPassword_requestButton"
            virtual={undefined}
          >
            <Styled(Text)
              virtual={undefined}
            >
              Request
            </Styled(Text)>
          </Styled(Button)>,
        ],
        "descriptionText": "Please enter the email address associated to your account to request a password reset. Check that email account for further instructions.",
        "headerText": "Forget your password?",
        "showBackButton": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dark": true,
            "leftIcon": "jbh-Contact_Mail",
            "onChangeText": [Function],
            "onValidationUpdate": [Function],
            "placeHolderText": "Email address",
            "style": Object {
              "marginTop": 44,
            },
            "value": "test@test.com",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": <Styled(Text)
              virtual={undefined}
            >
              Request
            </Styled(Text)>,
            "disabled": false,
            "large": true,
            "onPress": [Function],
            "primaryDark": true,
            "style": Object {
              "marginTop": 56,
            },
            "testID": "forgotPassword_requestButton",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Request",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Request",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
